device/cuda/src/cca/component_connection.cu

```diff
@@ -727,14 +724,16 @@ vecmem::vector<details::ccl_partition> partition(
 }
 }  // namespace details
 
+component_connection::component_connection(const traccc::memory_resource& mr)
+    : m_mr(mr) {}
+
 component_connection::output_type component_connection::operator()(
     const cell_container_types::host& data) const {
-    vecmem::cuda::managed_memory_resource upstream;
-    vecmem::binary_page_memory_resource mem(upstream);
+    vecmem::cuda::device_memory_resource dmr;
```

Comment:
We should really come to an agreement about what to do here. :thinking:

I'm still firmly in the belief that algorithms must not own their own memory resources, and must depend on outside resources to be given to them in their constructors. I'm okay with this algorithm not relying on a single `traccc::memory_resource` if you say that by definition it must not be used with a single shared/managed resource, but then make it require 2 resources all the time. Like:

```c++
component_connection(vecmem::memory_resource& host_mr, vecmem::memory_resource& device_mr);
```

But I can't give up on the exact memory resource "coming from the outside"...